src/osd/SnapMapper.cc

```diff
@@ -155,7 +155,10 @@ int SnapMapper::get_snaps(
     bufferlist::iterator bp = got.begin()->second.begin();
     decode(*out, bp);
     dout(20) << __func__ << " " << oid << " " << out->snaps << dendl;
-    assert(!out->snaps.empty());
+    if (out->snaps.empty()) {
+      dout(1) << __func__ << " " << oid << " empty snapset" << dendl;
+      assert(!cct->_conf->osd_debug_verify_snaps);
```

Comment:
no `return -ENOENT` here?

Reply:
Yes, it was intentional: we don't want `SnapMapper::get_snaps` to fail here, just return an empty snapset.